src/tike/operators/cupy/fresnelspectprop.py

```diff
@@ -0,0 +1,131 @@
+"""Defines a free-space propagation operator based on the CuPy FFT module."""
+
+__author__ = "Ashish Tripathi"
+__copyright__ = "Copyright (c) 2024, UChicago Argonne, LLC."
+
+import typing
+
+import numpy.typing as npt
+import numpy as np
+
+from .cache import CachedFFT
+from .operator import Operator
+
+
+class FresnelSpectProp(CachedFFT, Operator):
+    """Fresnel spectrum propagation (short range) using CuPy.
+
+    Take an (..., W, H) compelx array representing a wavefront and propagate.
+
+    Attributes
+    ----------
+    pixel_size : float
+        The realspace size of a pixel in meters
+    distance : float
+        The realspace propagation distance in meters
+    wavelength : float
+        The wavelength of the light in meters
```

Comment:
@a4894z, are these units correct? Are they all still correct if provided in meters?